ghost/email-service/lib/EmailEventStorage.js

```diff
@@ -128,11 +139,24 @@ class EmailEventStorage {
         }
     }
 
-    async unsubscribeFromNewsletters(event) {
+    async findNewslettersToKeep(event) {
         try {
-            await this.#membersRepository.update({newsletters: []}, {id: event.memberId});
+            const member = await this.#membersRepository.get({email: event.email}, {
+                withRelated: ['newsletters']
+            });
+            const existingNewsletters = member.related('newsletters');
+
+            if (!existingNewsletters) {
+                return [];
+            }
+
+            const email = await this.#models.Email.findOne({id: event.emailId});
+            const newsletterToRemove = await email.get('newsletter_id');
+
+            return existingNewsletters.toJSON().filter(newsletter => newsletter.id !== newsletterToRemove);
```

Comment:
I think we should still map this to ids, otherwise we do an update on a relation with lots of properties. Here or at least when using findNewslettersToKeep. It is probably also a bit more performant to use `models` instead of `toJSON` because we only need the ids later.
```suggestion
            return existingNewsletters.models.filter(newsletter => newsletter.id !== newsletterToRemove).map(n => { return {id: n.id}; });
```

Reply:
thanks! Updated in 35f0ab52b0bdc6db3da4e82e92a5f91338686728